File: 프로그래머스/lv3/43162. 네트워크/네트워크.java
class Solution {
    static boolean visited[];
    public int solution(int n, int[][] computers) {
        int answer = 0;
        
        visited = new boolean[n];
        
        for(int i = 0;i < n;i++) {
            if(!visited[i]) {
                answer++;
                dfs(i, n, computers);
            }
        }
        
        return answer;
    }
    
    public static void dfs(int i, int n, int[][] computers) {
        for(int j = 0;j < n;j++) {
            if(!visited[j] && computers[i][j] == 1) {
                visited[j] = true;
                dfs(j, n,computers);
            }
        }
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    let answer = 0;
    const visited = Array(n).fill(false)
    
    const dfs = (node) => {
        visited[node] = true
        
        for(let i = 0;i < n;i++) {
            if(!visited[i] && computers[node][i] === 1) dfs(i)
        }
    }
    
    for(let i = 0;i < n;i++) {
        if(visited[i]) continue
        
        dfs(i)
        answer += 1
    }
    
    return answer;
}